core/src/toga/icons.py

```diff
@@ -62,62 +62,89 @@ def OPTION_CONTAINER_DEFAULT_TAB_ICON(cls) -> Icon:
 
     def __init__(
         self,
-        path: str | Path,
+        path: str | Path | None,
         *,
         system: bool = False,  # Deliberately undocumented; for internal use only
+        default: (
+            toga.Icon | None
+        ) = None,  # Deliberately undocumented; for internal use only
     ):
         """Create a new icon.
 
         :param path: Base filename for the icon. The path can be an absolute file system
             path, or a path relative to the module that defines your Toga application
-            class.
-
-            This base filename should *not* contain an extension. If an extension is
-            specified, it will be ignored.
+            class. This base filename should *not* contain an extension. If an extension
+            is specified, it will be ignored. If :any:`None`, the application binary will
+            be used as the source of the icon.
```

Comment:
The comparison with Button/Table isn't entirely fair, because a Button or table item can have no icon at all; but an app *must* have an icon. However, I agree that `Icon.APP_ICON` is a more helpful representation to use instead of ``None``; this also gives us a way to hide a sentinel value so that we can construct `APP_ICON` without exposing a public constant or violate the type definition of the ``path`` argument.